### PR TITLE
FRDQ-115

### PR DESCRIPTION
**Pull request by hayuna**

### CODE REVIEW

File: src/Routes/Routes.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Route, Switch } from 'react-router-dom';
 import Spinner from 'components/atoms/Spinner';
 import { LoginPage } from 'Routes/helpers';
 import PrivateRoute from 'Routes/PrivateRoute';
+import ReduxTestComponent from '../redux/ReduxTestComponent/ReduxTestComponent';
```
Comment (Marcin-tech-ocean) [inline src/Routes/Routes.tsx:7]: use absolute path

File: src/redux/ReduxTestComponent/ReduxTestComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import loader from 'redux/reducers/loader';
+import { RootState, store } from 'redux/store';
+import { useSelector } from 'react-redux';
+import Spinner from '../../components/atoms/Spinner';
```
Comment (Marcin-tech-ocean) [inline src/redux/ReduxTestComponent/ReduxTestComponent.tsx:5]: use absolute path

File: src/redux/reducers/loader.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const initialState: { isLoading: boolean } = {
+  isLoading: false,
+};
+
+const loader = createSlice({
+  name: 'loader',
+  initialState,
+  reducers: {
+    show: (state): void => {
+      // eslint-disable-next-line no-param-reassign
```
Comment (Marcin-tech-ocean) [inline src/redux/reducers/loader.ts:12]: can we do something with it ?
https://eslint.org/docs/rules/no-param-reassign#props

**Review by Marcin-tech-ocean — CHANGES_REQUESTED** [review]

File: src/Routes/Routes.tsx (hunk 1 above)
Comment (Krysik) [inline src/Routes/Routes.tsx:7]: use absolute path

File: src/redux/ReduxTestComponent/ReduxTestComponent.tsx (hunk 2 above)
Comment (Krysik) [inline src/redux/ReduxTestComponent/ReduxTestComponent.tsx:5]: use absolute path

File: src/redux/ReduxTestComponent/ReduxTestComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const ReduxTestComponent = (): JSX.Element => {
+  const showLoader = useSelector((state: RootState) => state.loader.isLoading);
+  const { show, hide } = loader.actions;
+  return (
+    <>
+      {!showLoader ? null : <Spinner />}
+      <button type="button" onClick={() => store.dispatch(show())}>
+        Show loader
+      </button>
+      <button type="button" onClick={() => store.dispatch(hide())}>
```
Comment (Krysik) [inline src/redux/ReduxTestComponent/ReduxTestComponent.tsx:16]: I think that we should have a single function which toggle the loader ?

File: src/redux/reducers/loader.ts (hunk 3 above)
Comment (Krysik) [inline src/redux/reducers/loader.ts:12]: can we do something with it ?
https://eslint.org/docs/rules/no-param-reassign#props

File: src/redux/reducers/loader.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+const initialState: { isLoading: boolean } = {
+  isLoading: false,
+};
+
+const loader = createSlice({
+  name: 'loader',
+  initialState,
+  reducers: {
+    show: (state): void => {
```
Comment (Krysik) [inline src/redux/reducers/loader.ts:11]:
```javascript
toggleLoader: (state): void => {
  state.isLoading = !state.isLoading;
}
```

**Review by Krysik — CHANGES_REQUESTED** [review]

File: src/redux/reducers/loader.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      state.isLoading = true;
+    },
+    hide: (state): void => {
+      // eslint-disable-next-line no-param-reassign
+      state.isLoading = false;
+    },
+  },
+});
+
+export default loader;
```
Comment (hayuna) [inline src/redux/reducers/loader.ts:22]: On the outside we need only reducers, so you can export `loader.reducer`

File: src/redux/reducers/loader.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState: { isLoading: boolean } = {
```
Comment (hayuna) [inline src/redux/reducers/loader.ts:3]: Move this type to `type` / `interface`

File: src/redux/store.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { configureStore } from '@reduxjs/toolkit';
+import loader from 'redux/reducers/loader';
+
+export const store = configureStore({
+  reducer: {
+    loader: loader.reducer,
```
Comment (hayuna) [inline src/redux/store.ts:6]: https://github.com/Programmers-Only-Group/frontend/pull/119/files?file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#r505665488 then you could only use `loader` as exported one

File: src/redux/ReduxTestComponent/ReduxTestComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import { RootState, store } from 'redux/store';
+import { useSelector } from 'react-redux';
+import Spinner from '../../components/atoms/Spinner';
+
+const ReduxTestComponent = (): JSX.Element => {
+  const showLoader = useSelector((state: RootState) => state.loader.isLoading);
+  const { show, hide } = loader.actions;
+  return (
+    <>
+      {!showLoader ? null : <Spinner />}
```
Comment (hayuna) [inline src/redux/ReduxTestComponent/ReduxTestComponent.tsx:12]: This is the same :) 

```suggestion
      {!showLoader && <Spinner />}
```

File: src/redux/reducers/loader.ts (hunk 6 above)
Comment (TomaszKorenberg) [reply src/redux/reducers/loader.ts:17]: No, I'm using loader.reducer and loader.actions (in ReduxTestComponent component). The same way is in documentation.

**Pull request merged**